### ISSUE TITLE
Spam error message when creating issues/comments is not handled properly

### ISSUE THREAD

**Issue by halfbyte** [NARKOZ/gitlab#429]
The problem lies in [error.rb:62](https://github.com/NARKOZ/gitlab/blob/91afbd71df5fb8363bdd35e9ae620fc9a7cb3fcb/lib/gitlab/error.rb#L62) where the `handle_message()` method expects a multi-level hash/array construct whereas the spam error comes in as `{"error"=>"Spam detected"}`. The error message is:

```
NoMethodError: undefined method `join' for "Spam detected":String
    gitlab (4.6.1) lib/gitlab/error.rb:63:in `block in handle_message'
    gitlab (4.6.1) lib/gitlab/error.rb:62:in `map'
    gitlab (4.6.1) lib/gitlab/error.rb:62:in `handle_message'
    gitlab (4.6.1) lib/gitlab/error.rb:46:in `build_error_message'
    gitlab (4.6.1) lib/gitlab/error.rb:20:in `initialize'
    gitlab (4.6.1) lib/gitlab/request.rb:66:in `exception'
    gitlab (4.6.1) lib/gitlab/request.rb:66:in `raise'
    gitlab (4.6.1) lib/gitlab/request.rb:66:in `validate'
    gitlab (4.6.1) lib/gitlab/request.rb:46:in `block (2 levels) in <class:Request>'
    gitlab (4.6.1) lib/gitlab/client/issues.rb:56:in `create_issue'
```

In case you wonder how I found this problem: I'm testing our own API layer against Gitlab.com. I probably need to rethink this strategy and need to test it against a separate Gitlab instance :)

I'd be willing to provide a PR.

**Comment by NARKOZ** [comment]
> I'd be willing to provide a PR.

Please do.

**Comment by NARKOZ** [comment]
Fixed in version 4.7.0
